test/facilitator/approveBountyAmount.js

```diff
@@ -0,0 +1,234 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const chai = require('chai');
+const sinon = require('sinon');
+const Web3 = require('web3');
+const Facilitator = require('../../libs/Facilitator/Facilitator');
+
+const assert = chai.assert;
+
+describe('Facilitator.approveBountyAmount()', () => {
+  let facilitator;
+  let web3;
+  let gatewayAddress;
+  let coGatewayAddress;
+  let baseTokenAddress;
+  let facilitatorAddress;
+  let bountyAmount;
+
+  let mockBaseTokenContract;
+  let mockTx;
+  let spy;
+
+  const setup = function() {
+    // Mock facilitator.getBounty method to return expected bounty amount.
+    sinon.stub(facilitator, 'getBounty').callsFake(() => {
+      return bountyAmount;
+    });
+
+    // Mock facilitator.getBaseToken method to return expected base token address.
+    sinon.stub(facilitator, 'getBaseToken').callsFake(() => {
+      return baseTokenAddress;
+    });
+
+    // Mock an instance of BaseToken contract.
+    mockBaseTokenContract = sinon.mock(
+      facilitator.contracts.BaseToken(baseTokenAddress)
+    );
+    const baseTokenContract = mockBaseTokenContract.object;
+
+    // Mock approve transaction object.
+    mockTx = sinon.mock(
+      baseTokenContract.methods.approve(gatewayAddress, bountyAmount)
+    );
+    sinon.stub(mockTx.object, 'send').callsFake(() => {
+      return Promise.resolve({
+        account: gatewayAddress,
+        amount: bountyAmount
+      });
+    });
+
+    // Fake the approve call.
+    sinon.stub(baseTokenContract.methods, 'approve').callsFake(() => {
+      return mockTx.object;
+    });
+
+    // Fake the Gateway call to return gatewayContract object;
+    sinon.stub(facilitator.contracts, 'BaseToken').callsFake(() => {
+      return baseTokenContract;
+    });
+    //
+
+    sinon.stub(facilitator, 'sendTransaction').callsFake((tx, txOptions) => {
+      return new Promise(async function(resolve, reject) {
+        const sendResult = await tx.send();
+        resolve({ txResult: sendResult, txOptions: txOptions });
+      });
+    });
+
+    // Add spy on Facilitator.approveBountyAmount.
+    spy = sinon.spy(facilitator, 'approveBountyAmount');
+  };
+
+  const tearDown = function() {
+    // Restore all mocked and spy objects.
+    mockBaseTokenContract.restore();
+    mockTx.restore();
+    spy.restore();
+  };
+
+  beforeEach(() => {
+    // runs before each test in this block
+    web3 = new Web3(new Web3.providers.HttpProvider('http://127.0.0.1:9546'));
```

Comment:
Is this a running instance?

Reply:
Removed this from few places.
In StakeHelper tests, when I create a mock object, it expects the provider. So it is still there in few tests.